File: skypydb-js/src/database/operations/vector/embedding_function.ts
import type { EmbeddingFunction } from "../../../types";
import { DeleteCollectionMixin } from "./collections";

export abstract class EmbeddingFunctionMixin extends DeleteCollectionMixin {
  embedding_function: EmbeddingFunction | null = null;

  set_embedding_function(embedding_function: EmbeddingFunction | null): void {
    this.embedding_function = embedding_function;
  }
}

File: skypydb/database/operations/vector/embedding_function.py
"""Embedding function management for vector storage operations."""

from typing import Callable, List


class EmbeddingFunctionMixin:
    """Attach and replace the embedding function used by the vector backend."""

    def set_embedding_function(
        self,
        embedding_function: Callable[[List[str]], List[List[float]]],
    ) -> None:
        """Set the callable used to convert text into embedding vectors."""

        self.embedding_function = embedding_function
